app/models/blog_post.rb

```diff
@@ -34,7 +37,22 @@ class BlogPost < ApplicationRecord
 
   scope :order_newest_first, -> { order(created_at: :desc) }
 
+  def slug_candidates
+    [
+      :title,
+      %i[title id]
+    ]
+  end
+
   def featured?
     featured_blog_post.present?
   end
+
+  def should_generate_new_friendly_id?
+    title_changed?
+  end
+
+  def initialize_feature
+    featured_blog_post || build_featured_blog_post
```

Comment:
This method might seem a little strange but I needed it because of some quirks with `has_one`. If you use `build_association` on an object that already has an associated object, the associated object will be destroyed (like actually deleted from the database). This method allows getting the associated object if it exists or building it if not, which avoids that side effect.

Reply:
Just curious, Is there a particular reason why we're making this a has_one featured_blog_post?

Reply:
Oh, I see each blog can only be featured once

Reply:
I find this `initialize_feature` naming a bit confusing -- is it meant to create a feature or access a feature?
If the latter, it might be better to rename to `featured_post`, since it's returning the blog post. Or alternatively, perhaps overwrite the `attr_reader` method for `featured_blog_post` to access the associated blog post if exists, else build a new one.

As I write this out, I have some spidey-sense that's ringing re: feature envy within the BlogPost class -- seems like it may know a bit too much about this featured blog post model